src/main/scala/gestalt/Trees.scala

```diff
@@ -163,11 +162,19 @@ trait Trees extends Params with TypeParams with
   val TypeIdent: TypeIdentImpl
   trait TypeIdentImpl {
     def apply(name: String): TypeTree
+    def unapply(tpe: TypeTree): Option[String]
   }
 
   val TypeSelect: TypeSelectImpl
   trait TypeSelectImpl {
     def apply(qual: Tree, name: String): TypeTree
+    def unapply(tpe: TypeTree): Option[(Tree, String)]
+  }
+
+  val PathType: PathTypeImpl
+  trait PathTypeImpl {
+    def apply(qual: Option[Tree], name: String, targs: Seq[TypeTree]): TypeTree
+    def unapply(tpe: TypeTree) : Option[(Option[Tree], String, Seq[TypeTree])]
```

Comment:
Ah, I see. I agree, destructing of TypeTrees should be avoided if possible.

Destruct a type will pose some challenge. But it's good to see what kind of additional APIs we'll need to support this.

We may need extractors for `AppliedType` (defined in `TypeApplications`) and `TypeRef`.